### PR TITLE
Add energy validation UI

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Show validation errors/warnings. Bare UI while waiting for better design.

Requires https://github.com/home-assistant/core/pull/54567

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
This should also be added to the `energy-setup-wizard`

**Review by bramkragten — DISMISSED** [review]

**Comment by matthiasdebaat** [comment]
What does `not defined` mean and how can a user fix this? We can use an alert/warning bar, for example:

Configuration Energy alerts
https://www.figma.com/file/0yWK3l5paYugEgvp2GRsVN/Energy?node-id=598%3A4783

**Comment by ludeeus** [comment]
> What does `not defined` mean and how can a user fix this?

When you have defined an entity that does not exist, it's fixed by:
- Getting the integration that provides the entity to work again (in case of a broken integration)
- Replacing it with a valid existing entity (in case of a removed entity)

**Comment by matthiasdebaat** [comment]
Updated text
Configuration Energy alerts

**Comment by balloob** [comment]
Backend and UI updated to match design:

image

File: src/panels/config/energy/components/ha-energy-validation-message.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import { mdiAlertOutline } from "@mdi/js";
+import { css, html, LitElement } from "lit";
+import { customElement, property } from "lit/decorators";
+import { groupBy } from "../../../../common/util/group-by";
+import "../../../../components/ha-svg-icon";
+import { EnergyValidationIssue } from "../../../../data/energy";
+import { HomeAssistant } from "../../../../types";
+
+@customElement("ha-energy-validation-result")
```
Comment (bramkragten) [inline src/panels/config/energy/components/ha-energy-validation-message.ts:9]: Give the file (and class) the same name

**Review by bramkragten — DISMISSED** [review]

File: src/translations/en.json [hunk 2]
```diff
@@ -1079,6 +1080,42 @@
             "dialog": {
               "selected_stat_intro": "Select the entity that represents the device energy usage."
             }
+          },
+          "validation": {
```
Comment (balloob) [inline src/translations/en.json:1084]: @matthiasdebaat here are all the validation errors that we can show with their title/description.

File: src/translations/en.json [hunk 3]
```diff
@@ -1079,6 +1080,42 @@
...
+                "title": "Unexpected unit of measurement",
+                "description": "The following entities do not have expected units of measurement kWh or Wh:"
+              },
+              "entity_unexpected_unit_price": {
+                "title": "Unexpected unit of measurement",
+                "description": "The following entities do not have expected units of measurement that ends with /kWh or /Wh:"
+              },
+              "entity_unexpected_state_class_total_increasing": {
+                "title": "Unexpected state class",
+                "description": "The following entities do not have expeted state class \"total_increasing\""
```
Comment (balloob) [inline src/translations/en.json:1116]:
```suggestion
                "description": "The following entities do not have expected state class \"total_increasing\""
```

**Pull request merged**